src/server.rs

```diff
@@ -0,0 +1,145 @@
+use std::error::Error;
+use std::sync::Arc;
+
+use tokio::net::UdpSocket;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+use tokio::sync::Mutex;
+use tokio::task::{JoinError, JoinHandle};
+
+use crate::ScuttleButt;
+
+/// Buffer size for UDP messages.
+const BUF_SIZE: usize = 4096;
+
+/// UDP ScuttleButt server.
+pub struct ScuttleServer {
+    pub scuttlebutt: Arc<Mutex<ScuttleButt>>,
```

Comment:
I am a bit scared by the idea of exposing this `Arc<Mutex>` as public.
On the other hand I am not entirely sure how to fix it (copying accessor, per node lock, concurrent hashmap, ...) without us knowing a little more about its usage.